Tests/LoggingTests/LoggingTest.swift

```diff
@@ -723,6 +723,25 @@ class LoggingTest: XCTestCase {
         XCTAssert(interceptStream.strings[1].contains("a=a1 b=b1"))
     }
 
+    func testStdioOutputStreamWrite() {
+        self.withWriteReadFDsAndReadBuffer { writeFD, readFD, readBuffer in
+            let logStream = StdioOutputStream(file: writeFD, flushMode: .always)
+            LoggingSystem.bootstrapInternal { StreamLogHandler(label: $0, stream: logStream) }
+            let log = Logger(label: "test")
+            let testString = "hello\u{0} world"
+            log.critical("\(testString)")
+
+            read(readFD, readBuffer, 256)
+
+            if let output = String(validatingUTF8: readBuffer) {
```

Comment:
I think a better way to write this would be

```swift
if let output = String(decoding: UnsafeRawBufferPointer(start: UnsafeRawPointer(readBuffer), count: returnValueOfRead), as: UTF8.self) {
```

because currently, you're relying on the initialisation of the read buffer.